### DISCUSSION TITLE
Terminal does not remember state on reload

### DISCUSSION THREAD

**Discussion by nooobcoder** [coder/code-server#2793] [Ideas]
**JUST FOR INFORMATION** - I am aware of some wayarounds like, **Screen** but I need a simple way. The terminal should persist it's state. Either it should kill the processes on disconnection from client or it should hold it's state anyhow for some time before being killed.

_I know there is a long discussion does already exist, but this one is different, as I have also included the problem of the **extensions reloading on window reload**._

1) It has been a requested feature since a long time, **the terminal in code-server forgets it's state on reload**. It is a headache in Node.js development when we develop a web-server. The port is allocated (the process keeps running in background) when the window is reloaded but the terminal forgets it's last state. It would be better if the code-server remembers it's state even after the window is reloaded.

2) Also, **the extensions reload every time when the window is refreshed.** 

These two features are needed very urgently. @code-asher @bpmct please help me if there is already a way that I am not aware of.

While developing React apps, i face these issues, that the port is being used in the background and the terminal forgets it's state on window reload.

**Comment by code-asher** [comment]
Persistent terminal progress can be tracked here: https://github.com/cdr/code-server/issues/2276

But there are currently no plans on persisting the extension host. I definitely agree that persisting would be superior but this would probably need to be implemented upstream.
